experiments/customui/parent.js

```diff
@@ -250,30 +250,77 @@ var ex_customui = class extends ExtensionCommon.ExtensionAPI {
       return result;
     };
 
-    // Enables dynamic height and fixed 100% width for a WebExtension frame
-    const setWebextFrameSizesForVerticalBox = function(frame, options) {
-      frame.width = "100%";
-      frame.height = (options.height || 100) + "px";
-      frame.style.display = options.hidden ? "none" : "block";
+    // Set a dimension of the frame given in options
+    //  and enable changing by CustomUILocalOptionsListener
+    // If dimension it is not provided in options (a number from API call, in pixels),
+    //  set default value: defaultValue (must include the unit)
+    // dimensionName: "width" or "height"
+    // Only to be called by setWebextFrameDimension
+    const setWebextFrameDynamicDimension = function(frame, options, dimensionName, defaultValue) {
+      frame[dimensionName] = (options[dimensionName] + "px" || defaultValue);
```

Comment:
This is evaluating to `"nullpx"` if the option is falsy.

Reply:
Nice catch! As loading the previous width is now working in my addon (thanks for the pointers!), I was indeed not testing the default value anymore, sorry.